packages/acs-ui-common/src/conditional-compilation-sample/index.tsx

```diff
@@ -175,6 +175,39 @@ function propTrampoline(): string {
  *
  */
 
+/**
+ * Conditional compilation directives can refer to one or more predefined features.
+ *
+ * When working on a beta feature, define a new feature tag in common/config/babel/.babelrc.js
+ * Then use that tag in all the conditional compilation directives for your feature.
+ *
+ * Most examples in this file use the `demo` feature.
+ *
+ * When your feature is ready for stable builds: Move the feature flag to the `stabilizedFeatures` list.
+ * The tagged code will no longer be removed from stable builds.
+ *
+ * Using an undeclared feature in conditional compilation directive will cause a build failure (prevents typos).
+ *
+ * The following code is not removed because it contains a stabilized feature tag.
+ */
+/* @conditional-compile-remove(stabilizedDemo) */
+export const thisIsNowStable = 42;
+
+/* The following is not removed because it contains one stabilized feature, along with a beta-only feature.
```

Comment:
What would a good situation to use this stack of conditional compile features be? I see what its saying but I am wondering if there is a good example of a use case?

Reply:
This is used currently in places by the camera switcher button and control bar split buttons.
That is an example of smaller features that add up to a larger feature -- we could decide to stabilize the smaller features first (just release the split buttons... because they get used elsewhere) or release the bigger feature with everything involved.

It's still up for debate whether that's a good idea or not.


Another example (doesn't exist yet) is if a field is needed by two somewhat independent features (e.g., and `options` extension to some existing API, where individual features add different fields to the options).

Reply:
Oh I see! so like how my camera switcher is dependent on the CallWithChatComposite to go live so it might still be in beta while the CallWithChatComposite is stabalized. I like it!